test/linter/test-versions.js

```diff
@@ -111,59 +145,60 @@ function addedBeforeRemoved(statement) {
   const browsersToCheck = Object.keys(supportData);
+
   for (const browser of browsersToCheck) {
     if (validBrowserVersions[browser]) {
       /** @type {SimpleSupportStatement[]} */
-      const supportStatements = [];
-      if (Array.isArray(supportData[browser])) {
-        Array.prototype.push.apply(supportStatements, supportData[browser]);
-      } else {
-        supportStatements.push(supportData[browser]);
-      }
-
-      const validBrowserVersionsString = `true, false, null, ${validBrowserVersions[
-        browser
-      ].join(', ')}`;
-      const validBrowserVersionsTruthy = `true, ${validBrowserVersions[
-        browser
-      ].join(', ')}`;
+      const supportStatements = Array.isArray(supportData[browser])
+        ? supportData[browser]
+        : [supportData[browser]];
 
       let sawVersionAddedOnly = false;
 
       for (const statement of supportStatements) {
-        for (const property of ['version_added', 'version_removed']) {
-          if (!isValidVersion(browser, statement[property])) {
+        if (statement === undefined) {
+          if (blockList[category].includes(browser)) {
             logger.error(
-              chalk`{red → {bold ${relPath}} - {bold ${property}: "${
-                statement[property]
-              }"} is {bold NOT} a valid version number for {bold ${browser}}\n    Valid {bold ${browser}} versions are: ${validBrowserVersionsString}}${
-                browserTips[browser]
-                  ? chalk`\n    {blue {bold Tip:} ${browserTips[browser]}}`
-                  : ''
-              }`,
+              chalk`{red → {bold ${browser}} must be defined for {bold ${relPath}}}`,
             );
           }
-        }
-
-        if ('version_added' in statement && 'version_removed' in statement) {
-          if (statement.version_added === statement.version_removed) {
-            logger.error(
-              chalk`{red → {bold ${relPath}} - {bold version_added: "${statement.version_added}"} must not be the same as {bold version_removed} for {bold ${browser}}}`,
-            );
+        } else {
+          for (const property of ['version_added', 'version_removed']) {
+            if (
+              property == 'version_removed' &&
+              statement.version_removed === undefined
+            ) {
+              // Undefined is allowed for version_removed
+              continue;
+            }
+            if (!isValidVersion(browser, category, statement[property])) {
+              logger.error(
+                chalk`{red → {bold ${relPath}} - {bold ${property}: "${
+                  statement[property]
+                }"} is {bold NOT} a valid version number for {bold ${browser}}\n    Valid {bold ${browser}} versions are: ${validBrowserVersions[
+                  browser
+                ].join(', ')}}${
+                  browserTips[browser]
+                    ? chalk`\n    {blue {bold Tip:} ${browserTips[browser]}}`
+                    : ''
+                }`,
```

Comment:
Would it make sense to pass the tip as the 2nd parameter, like you did in other PRs?

```suggestion
                ].join(', ')}}`,
                browserTips[browser],
```